### PR TITLE
Implement About Window For Linux

### PR DESCRIPTION
**Pull request by xyloflake**
Until now, there was no option to show the about window on linux. Now, users can open the about window by right clicking on the application in the dock. This is done by setting up custom actions in the .desktop file.

### CODE REVIEW

**Comment by xyloflake** [comment]
Do NOT merge without proper testing.

File: electron-builder.json
Comment (xyloflake) [inline electron-builder.json:21]: Updated category

File: electron-builder.json
Comment (xyloflake) [inline electron-builder.json:34]: These are added to add the custom action at the end, otherwise electron-builder would add these at the end

File: electron-builder.json
Comment (xyloflake) [inline electron-builder.json:35]: This looks weird because of the hacky workaround that was needed to make it work as electron-builder does not have support for custom actions in the .desktop file

File: electron-builder.json
Comment (xyloflake) [inline electron-builder.json:51]: removed trailing comma

**Comment by xyloflake** [comment]
I've explained the code changes in the review comments.

**Comment by staniel359** [comment]
@xyloflake What about translations?

**Comment by xyloflake** [comment]
> @xyloflake What about translations?

Ignore. "About" doesn't change in languages.

**Comment by staniel359** [comment]
@xyloflake What do you mean?

**Comment by xyloflake** [comment]
> @xyloflake What do you mean?

I mean that translations are not important for now.

File: electron/actions/app/setup.js
Comment (xyloflake) [inline electron/actions/app/setup.js:23]: @gamersi do we still need this even after you handled it in `handleSecondInstance`?

Comment (xyloflake) [reply electron/actions/app/setup.js:23]: Maybe you'd need that when it's NOT a second instance?

**Comment by xyloflake** [comment]
@staniel359 @gamersi 

1. Let's add the translations actually
2. Let's merge this after a beta test (for obv reasons)
3. Awaiting testing on KDE

**Comment by xyloflake** [comment]
@staniel359 @gamersi I have come to the realisation that changing the "about" for different languages is NOT possible.

Here's why:
1. You'd need to change the .desktop file when the language is changed. Changing the .desktop file is NOT possible at runtime and even if we manage to change it, it'd require a DE reload. It'd be bad UX if the effect is not immediately seen.
2. Changing about to other languages should be prevented because we can't rely on the DE to correctly render the languages. For example, we have Hebrew, which I'm pretty sure would end up messed because gnome and KDE have some weird behaviour with those languages.
3. .desktop is generated at build time. Therefore to modify it at runtime without sudo would be IMPOSSIBLE. And knowing what language the user would use at build time is impossible as well.

We should let "about" be about. At least for the time being until the autoupdates are fixed on Pacman.

File: electron/actions/app/setup.js
Comment (gamersi) [reply electron/actions/app/setup.js:23]: Yes

**Comment by gamersi** [comment]
> @staniel359 @gamersi I have come to the realisation that changing the "about" for different languages is NOT possible.
> 
> Here's why:
> 1. You'd need to change the .desktop file when the language is changed. Changing the .desktop file is NOT possible at runtime and even if we manage to change it, it'd require a DE reload. It'd be bad UX if the effect is not immediately seen.
> 2. Changing about to other languages should be prevented because we can't rely on the DE to correctly render the languages. For example, we have Hebrew, which I'm pretty sure would end up messed because gnome and KDE have some weird behaviour with those languages.
> 3. .desktop is generated at build time. Therefore to modify it at runtime without sudo would be IMPOSSIBLE. And knowing what language the user would use at build time is impossible as well.
> 
> We should let "about" be about. At least for the time being until the autoupdates are fixed on Pacman.

Correct. I mean we *could* change it (with sudo) but it's not meant to be and if stuff could break.

**Review by staniel359 — DISMISSED** [review]
LGTM

**Comment by xyloflake** [comment]
Disabled auto merge. We will only merge after the beta test for obvious reasons.

@gamersi please report the status for KDE.

**Comment by xyloflake** [comment]
Beta test has started folks.

Cheers 🍻

**Comment by xyloflake** [comment]
[Screencast from 2024-07-30 19-05-16.webm](https://github.com/user-attachments/assets/3021e609-4b7b-4808-a5ae-667ab5595c8b)

Works on my end.

File: electron/handlers/app.js
Comment (xyloflake) [inline electron/handlers/app.js:52]: Because mainWindow should have no behaviour when the action is to open the about window.

**Review by xyloflake — COMMENTED** [review]
LGTM, ready to be merged.

**Review by staniel359 — APPROVED** [review]

**Comment by xyloflake** [comment]
As has been reported by our beta testers; the implementation works correctly. Therefore, enabling auto merge as we're ready for this to be pushed.

**Comment by gamersi** [comment]
> Disabled auto merge. We will only merge after the beta test for obvious reasons.
> 
> @gamersi please report the status for KDE.

Works on KDE as stated on discord

**Review by gamersi — APPROVED** [review]
LGTM

**Comment by xyloflake** [comment]
Our first coauthor together @staniel359 @gamersi 🚀

Nice to work with you guys as always. This felt a bit hurried and it was the fastest we've ever pushed out a feature lol.

**Pull request merged**
